src/test/test-restrict-ifaces.c

```diff
@@ -0,0 +1,264 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <stdlib.h>
+
+#include "load-fragment.h"
+#include "manager.h"
+#include "process-util.h"
+#include "restrict-ifaces.h"
+#include "rm-rf.h"
+#include "service.h"
+#include "strv.h"
+#include "tests.h"
+#include "unit.h"
+#include "virt.h"
+
+/* create_veth_pairs creates n network namespaces and n veth pairs. One
+ * peer of each veth pair is moved to a different network namespace.
+ * Each veth pair forms a /30 network, the first address of the subnet
+ * is assigned to the peer in the network namespace and the next one to
+ * the peer on the host network namespace. Pinging the address of the
+ * peer in the network namespace forces the packets to pass through
+ * the corresponding veth peer in the host network namespace.
+ *
+ * The IP address used to ping in this test are the following ones:
+ * - basenetwork.1 -> veth0
+ * - basenetwork.5 -> veth1
+ * - basenetwork.9 -> veth2
+ * - basenetwork.${4*N+1} -> veth${N}
+ */
+
+static int create_veth_pairs(int nveths, const char *basenetwork) {
+        int r;
+        char cmd[1024];
+        const char setup_veth[] = "#! /bin/sh\n"
+                                "set -euxo\n"
+                                "for i in `seq 0 %d`;\n"
+                                "do\n"
+                                "    ip netns del ns${i} || true\n"
+                                "    ip link del veth${i} || true\n"
+                                "    ip netns add ns${i}\n"
+                                "    ip link add veth${i} type veth peer name veth${i}_\n"
+                                "    ip link set veth${i}_ netns ns${i}\n"
+                                "    ip -n ns${i} link set dev veth${i}_ up\n"
+                                "    ip -n ns${i} link set dev lo up\n"
+                                "    ip -n ns${i} addr add %s$((4*i+1))/30 dev veth${i}_\n"
+                                "    ip link set dev veth${i} up\n"
+                                "    ip addr add %s$((4*i+2))/30 dev veth${i}\n"
+                                "done";
+        r = snprintf(cmd, sizeof(cmd), setup_veth, nveths-1, basenetwork, basenetwork);
+        if ((size_t) r >= sizeof(cmd))
+                return -1;
+        r = system(cmd);
+        if (r < 0)
+                return r;
+        if (WEXITSTATUS(r) != 0)
+                return -1;
+        return 0;
+}
```

Comment:
humm... this is more an integration test. please use the test/TEST-XY… testsuite logic for these.

Reply:
I updated it but they are failing. This feature needs kernel 5.7 and I think we're not running any integration test with that version. Any suggestion how to proceed?